lecture2gether_flask/app.py

```diff
@@ -126,14 +126,18 @@ def decode_l2go_path():
     try:
         if url.hostname in ['www.youtube.com', 'youtube.com', 'youtu.be']:
             meta_data_provider = YouTubeMetaDataProvider(video_url)
+        elif url.hostname in ['drive.google.com', ]:
+            meta_data_provider = GoogleDriveMetaDataProvider(video_url)
         elif 'lecture2go' in url.hostname or '/l2go/' in url.path:
             meta_data_provider = L2GoMetaDataProvider(video_url, password)
         else:
             meta_data_provider = DefaultMetaDataProvider(video_url)
     except VideoNotFoundException:
-        abort(404)
+        abort(404, "Video not found")
     except VideoUnauthorizedException:
-        abort(401)
+        abort(401, "You are not authorized to watch this video")
+    except APIUnauthorizedException:
+        abort(501, "Lecture2Gether needs a working Google API key for this feature!")
```

Comment:
Maybe a little bit more generic like: "L2Ge cannot access necessary API. Is the API key (e.g. for Google) correct?"

Reply:
Oka

Reply:
But we don't need the API key to run l2g. I only want to tell, that this feature is only available with an API key (only the google stuff at the moment)

Reply:
The key is not necessary to run l2ge, but it is required for this feature... so maybe:
"Lecture2Gether cannot access required API to access data. Is the API key (e.g. for Google) correct?"

Reply:
Okay